zokrates_core/src/proof_system/libsnark/ffi.rs

```diff
@@ -20,7 +20,7 @@ extern "C" {
 }
 
 impl Buffer {
-    pub fn from_vec(v: &Vec<u8>) -> Buffer {
+    pub unsafe fn from_vec(v: &Vec<u8>) -> Buffer {
```

Comment:
This function is unsafe as it itentionally leaves a memory leak which must be handled by the caller. I marked it unsafe to prevent any future usage outside of the unsafe block.

Reply:
Interesting. I never really thought about this too much, but my assumption is that `unsafe` was exclusively to turn off some compiler safety checks, so as a result, anything that would compile without `unsafe` shouldn't have the flag. Does this function compile without `unsafe`?

Reply:
Yes you can compile this without the `unsafe` keyword, though after further digging through the rust documentation, it looks like Rust considers memory leaks to be safe. I'll remove this for now.